src/nanoarrow/common/inline_array.h

```diff
@@ -752,19 +763,43 @@ static inline ArrowErrorCode ArrowArrayFinishElement(struct ArrowArray* array) {
 
   switch (private_data->storage_type) {
     case NANOARROW_TYPE_LIST:
-    case NANOARROW_TYPE_MAP:
+    case NANOARROW_TYPE_LIST_VIEW:
+    case NANOARROW_TYPE_MAP: {
       child_length = array->children[0]->length;
       if (child_length > INT32_MAX) {
         return EOVERFLOW;
       }
       NANOARROW_RETURN_NOT_OK(
           ArrowBufferAppendInt32(ArrowArrayBuffer(array, 1), (int32_t)child_length));
+
+      if (private_data->storage_type == NANOARROW_TYPE_LIST_VIEW) {
+        struct ArrowBufferView buf_view;
+        buf_view.data.data = ArrowArrayBuffer(array, 1)->data;
+        const int32_t array_len = array->length;
+        const int32_t prev_offset =
+            array_len > 0 ? buf_view.data.as_int32[array_len - 1] : 0;
+        NANOARROW_RETURN_NOT_OK(ArrowBufferAppendInt32(
+            ArrowArrayBuffer(array, 2), (int32_t)child_length - prev_offset));
+      }
       break;
+    }
     case NANOARROW_TYPE_LARGE_LIST:
+    case NANOARROW_TYPE_LARGE_LIST_VIEW: {
       child_length = array->children[0]->length;
       NANOARROW_RETURN_NOT_OK(
           ArrowBufferAppendInt64(ArrowArrayBuffer(array, 1), child_length));
+      if (private_data->storage_type == NANOARROW_TYPE_LARGE_LIST_VIEW) {
```

Comment:
Generally not sure how much we want the implementation of the list views tied to their respective plain list types. Here I've kept the cases group together and specialize within the branch, but also happy to move the view types to their own case statements